packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -342,12 +352,16 @@ class DatePicker extends React.Component {
     );
 
     return (
-      <div className={styles['date-picker']}>
-        <ResponsiveElement
-          responsiveTo="window"
-          defaultElement={portalPicker}
-          small={popupPicker}
-        />
+      <div className={cx(['container-div'])}>
+        {required && <span className={cx(['required'])}>*</span>}
```

Comment:
Have you checked with UX that adding an asterisk to the front of the date input is how it should look like for required? This is just the date input instead of a date field as in https://engineering.cerner.com/terra-core/#/tests/terra-form-field/form-field/field-label-required

Reply:
Since the terra-date-picker doesn't have a default label, anyone that uses it can add a label and then wrap both of those elements int a terra-form-field to get the required asterisk next to the label. My understanding of #470 is that consumers wanted to be able to add a required asterisk by just passing in a prop

Reply:
Should we be directly add this to the date-pick? Or should we create a new component called date-picker-field which uses terra-form-field to wrap the date picker? That would give the date-picker access styling, etc that form-field provides.

For example, what about the error text that may be associated with the required field?

Reply:
I would be fine honestly with leaving it off for now, and waiting for when we add a field component to this for the rendering of the *. The key here would be making sure the inputs are getting their correct attributes.

Reply:
I'm also fine with leaving out the rendering of the * and just set the required attribute...just like what Input does today. I just think rendering the * directly in front of the date input is inconsistent with all other form fields where the * is rendered in front of the label.

Reply:
Talked with our UX designer @neilpfeiffer. The preference from UX is to leaving the asterisk out of the terra-datepicker component. This will be added in a a terra-datepicker-field component we create instead. 

For this PR, I agree with the others, let's leave the required prop and functionality of adding the aria-required and just drop the rendering of the asterisk.